codalab/apps/web/static/js/worksheet/worksheet_interface.js

```diff
@@ -148,28 +152,26 @@ var Worksheet = React.createClass({
           $("#raw-textarea").focus();
         }
     },
-    toggleActionBar: function(){
-        this.setState({showActionBar:!this.state.showActionBar});
+    toggleActionBar: function() {
+        this.setState({showActionBar: !this.state.showActionBar});
     },
-    showActionBar: function(){
-        this.setState({showActionBar:true});
+    showActionBar: function() {
+        this.setState({showActionBar: true});
     },
-    hideActionBar: function(){
-        this.setState({showActionBar:false});
+    hideActionBar: function() {
+        this.setState({showActionBar: false});
     },
-    refreshWorksheet: function(){
+    refreshWorksheet: function() {
         $('#update_progress').show();
-        this.setState({updating:true});
+        this.setState({updating: true});
         ws_obj.fetch({
-            success: function(data){
-                // console.log("fetch_and_update success");
-                // console.log("%c--------------------------------------------------------------", "color: Green; font-size:15px;");
-                if(this.isMounted()){
+            success: function(data) {
+                if (this.isMounted()) {
                     this.refs.list.setState({worksheet: ws_obj.getState()});
```

Comment:
This may be an anti pattern for React. If we are setting the state of anything other than self in a component's class, that indicates that the source data is not stored at one place and we have multiple copies which is not ideal when the virtual DOM is created and the diffing happens. 

However, to fix this, we would have to do some major re-factoring of the code.